### ISSUE TITLE
Sqllite caching backend broken in 0.5.* releases

### ISSUE THREAD

**Issue by ryanohnemus** [stevearc/pypicloud#119]
I am using the sqllite caching backend with pypicloud. When moving from version 0.4.5 to 0.5.0 and rebuilding the cache (by hitting the rebuild button within the web ui, or by just completely removing the db.sqllite file and restarting) I am unable to upload any packages or view them using the `http://_pypicloud_ip_/pypi` url.

This is what is returned from `http://_pypicloud_ip_/pypi`:
```xml

<?xml version='1.0'?>
<methodResponse>
<fault>
<value><struct>
<member>
<name>faultCode</name>
<value><int>-32700</int></value>
</member>
<member>
<name>faultString</name>
<value><string>parse error; not well formed</string></value>
</member>
</struct></value>
</fault>
</methodResponse>
```

Turning on DEBUG for the root logger I  get the following when trying to either upload a package or hit http://_pypicloud_ip_/pypi:
```
[Tue Apr 18 08:42:17 2017] POST /pypi/ => generated 288 bytes in 160 msecs (HTTP/1.1 200) 2 headers in 79 bytes (1 switches on core 0)
DEBUG 2017-04-18 08:42:36,020 [txn.140016947050432] new transaction
DEBUG 2017-04-18 08:42:36,026 [txn.140016947050432] abort
```

Falling back to 0.4.6 and rebuilding the cache again allows me to upload packages and see them in the UI and this is what is printed in the log
```
DEBUG 2017-04-18 08:50:15,546 [txn.140349990668224] new transaction
DEBUG 2017-04-18 08:50:15,550 [pypicloud] no route matched for url http://_pypicloud_ip_/pypi/
DEBUG 2017-04-18 08:50:15,629 [txn.140349990668224] commit <zope.sqlalchemy.datamanager.SessionDataManager object at 0x7fa5bf575d90>
DEBUG 2017-04-18 08:50:15,630 [txn.140349990668224] commit
```

**Comment by triplepoint** [comment]
Can confirm I'm seeing the same behavior with a stock install of the 0.5.1 docker image.

**Comment by stevearc** [comment]
Thanks for the report! I'll try to look into it this weekend

**Comment by benh57** [comment]
I'm actually seeing the same behavior with the redis backend. I think uploading might just be totally broken with 0.5.1, will roll back to earlier.

**Comment by benh57** [comment]
Verified  (with redis) - downgraded to 0.4.5, regenerated index, upload is functional.

Upgraded to 0.5.0, Regenerated index in GUI, upload failed with 'abort' log:

DEBUG 2017-04-19 22:40:08,233 [txn.140524273960704] commit
DEBUG 2017-04-19 22:40:08,238 [txn.140524282353408] new transaction
DEBUG 2017-04-19 22:40:08,291 [txn.140524282353408] commit

**Comment by stevearc** [comment]
Okay, I believe I have a partial fix coming. The long-term fix should be to replace `/pypi` with `/simple` in your urls. It was my mistake to ever use `/pypi` in the documentation. It's strictly incorrect, but it didn't matter until we added the XMLRPC search endpoint at `/pypi`

**Comment by stevearc** [comment]
Actually I think I have a full fix that won't require any url change. You can of course apply the url change now as a workaround.

**Comment by stevearc** [comment]
Okay, I released [0.5.2](https://pypi.python.org/pypi/pypicloud/0.5.2) which should fix these issues. Let me know if they persist.

**Comment by ryanohnemus** [comment]
Thank you for the fix! Looks good!
